core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/CoarseMesosSchedulerBackend.scala

```diff
@@ -152,6 +164,15 @@ private[spark] class CoarseMesosSchedulerBackend(
     command.build()
   }
 
+  protected def driverURL: String = {
```

Comment:
a little strange that this is protected. Can you just make it private and introduce an if case for the tests:
```
val testing = conf.get("spark.testing", "false").toBoolean
if (testing) {
  "driverURL"
} else {
  AkkaUtils.address(...)
}
```